src/path.rs

```diff
@@ -0,0 +1,63 @@
+use pathfinding::prelude::bfs;
+
+use crate::grid::Grid;
+
+#[derive(Clone, Debug, Eq, Hash, Ord, PartialEq, PartialOrd)]
+pub struct Pos(i32, i32);
+
+impl Pos {
+    fn successors(&self) -> Vec<Pos> {
+        let &Pos(x, y) = self;
+        vec![Pos(x + 1, y), Pos(x - 1, y), Pos(x, y + 1), Pos(x, y - 1)]
+    }
+}
+
+impl From<&Pos> for (i32, i32) {
+    fn from(val: &Pos) -> Self {
+        (val.0, val.1)
+    }
+}
+
+pub fn resolve(grid: &Grid, goals: &[(i32, i32)]) -> Option<Vec<(i32, i32)>> {
+    let mut res = vec![];
+    for i in 1..goals.len() {
+        let prev_goal = goals[i - 1];
+        let goal = goals[i];
+        let to_add = bfs(
+            &prev_goal,
+            |p| {
+                let mut succ = grid.neighbours(*p);
+                succ.retain(|x| match x {
+                    // FIXME: map borders exist to avoid infinite search.
```

Comment:
See https://discord.com/channels/691052431525675048/891721828123414538/891808675512090635

Reply:
Yeah I'll settle on map bounds eventually, this is a perfectly fine temporary measure